src/plugins/gui/pluginUtils.ts

```diff
@@ -48,32 +39,53 @@ export const getBestError = (errorQuotes: FiatProviderError[], currencyCode: str
       bestError = errorQuote
       continue
     }
-    // @ts-expect-error
     if (ERROR_PRIORITIES[errorQuote.errorType] < ERROR_PRIORITIES[bestError.errorType]) {
       bestError = errorQuote
       continue
     }
-    // @ts-expect-error
     if (ERROR_PRIORITIES[errorQuote.errorType] === ERROR_PRIORITIES[bestError.errorType]) {
       if (errorQuote.errorType === 'overLimit' && bestError.errorType === 'overLimit') {
-        if (errorQuote.errorAmount > bestError.errorAmount) {
+        if (bestError.errorAmount != null && errorQuote.errorAmount != null && errorQuote.errorAmount > bestError.errorAmount) {
           bestError = errorQuote
         }
       } else if (errorQuote.errorType === 'underLimit' && bestError.errorType === 'underLimit') {
-        if (errorQuote.errorAmount < bestError.errorAmount) {
+        if (bestError.errorAmount != null && errorQuote.errorAmount != null && errorQuote.errorAmount < bestError.errorAmount) {
           bestError = errorQuote
         }
       }
     }
   }
   if (bestError == null) return
-  let errorText = ERROR_TEXT[bestError.errorType]
-  if (bestError.errorType === 'underLimit' || bestError.errorType === 'overLimit') {
-    const localeAmount = formatNumber(bestError.errorAmount.toString())
-    errorText = sprintf(errorText, localeAmount + ' ' + currencyCode)
-  } else if (bestError.errorType === 'regionRestricted') {
-    errorText = sprintf(errorText, bestError.displayCurrencyCode)
+  return getErrorText(bestError, currencyCode)
+}
+
+const getErrorText = (error: FiatProviderQuoteError, currencyCode: string): string => {
+  let errorText = ''
+  const errorAmount =
+    (error.errorType === 'overLimit' || error.errorType === 'underLimit') && error.errorAmount != null
```

Comment:
Do not need to check the type. Only `overLimit` and `underLimit` can have `errorAmount`

Reply:
The way we have the types set up we need to check the type. Lint prevents usage of error.hasOwnProperty('errorAmount').

Reply:
While that's true, the way the types are currently set up, we can't check errorAmount here without first directly
checking errorType against both string literals. I may have addressed this issue
with my original error type changes that were just reverted but will note this
in https://app.asana.com/0/1200382638405084/1204654536856719/f

Duplicated some code in the switch cases instead for now.